File: JAVA/PokerWorld/src/bluffinmuffin/poker/game/IPokerGame.java
package bluffinmuffin.poker.game;

import bluffinmuffin.poker.game.observer.IPokerGameListener;

public interface IPokerGame {

    /**
     * Debute l'ecoute de tous les evenements de la Game
     * 
     * @param listener
     */
    void attach(IPokerGameListener listener);

    /**
     * Termine l'ecoute de tous les evenements de la Game
     * 
     * @param listener
     */
    void detach(IPokerGameListener listener);

    /**
     * La table associee a la game
     * 
     * @return
     */
    TableInfo getTable();

    /**
     * L'argent jouee par un joueur precis
     * 
     * @param player
     * @param amount
     * @return
     */
    boolean playMoney(PlayerInfo player, int amount);

    /**
     * Un joueur precis quitte la Game
     * 
     * @param player
     * @return
     */
    boolean leaveGame(PlayerInfo player);

    public String encode();
}

File: C#/BluffinMuffin.Poker.DataTypes/IPokerGame.cs
using BluffinMuffin.Poker.DataTypes.EventHandling;

namespace BluffinMuffin.Poker.DataTypes
{
    public interface IPokerGame
    {
        PokerGameObserver Observer { get; }

        TableInfo Table { get; }

        bool PlayMoney(PlayerInfo p, int amnt);
        int AfterPlayerSat(PlayerInfo p, int noSeat = -1, int moneyAmount = 1500);
        bool SitOut(PlayerInfo p);

        bool IsPlaying { get; }
    }
}
